simapp/v2/simdv2/cmd/commands.go

```diff
@@ -70,17 +68,26 @@ func initRootCmd[T transaction.Tx](
 		offchain.OffChain(),
 	)
 
+	storeComponent := store.New[T]()
 	// wire server commands
 	if err = serverv2.AddCommands(
 		rootCmd,
 		newApp,
 		logger,
 		cometbft.New(&genericTxDecoder[T]{txConfig}, cometbft.DefaultServerOptions[T]()),
 		grpc.New[T](),
-		store.New[T](),
+		storeComponent,
 	); err != nil {
 		panic(err)
 	}
+
+	// Add RestoreSnapshotCmd separately cause need appCreator
```

Comment:
imho we still need `appCreator` here for `storeKeys` since removing ss & sc database for restore cmd

Reply:
hum, cannot `store.New` take the appCreator?